library/ingest.py

```diff
@@ -38,153 +19,150 @@ def compress(self, path: str, *files, inplace: bool = True):
+                fields=destination["fields"],
+            )
+
+            # Create output folder and output config
+            if folder_path and output_suffix:
+                os.makedirs(folder_path, exist_ok=True)
+                self.write_config(f"{folder_path}/config.json", c.compute_json)
+                self.write_config(f"{folder_path}/config.yml", c.compute_yml)
+
+            # Initiate vector translate
+            gdal.VectorTranslate(
+                dstDS,
+                srcDS,
+                format=output_format,
+                layerCreationOptions=destination["options"],
+                dstSRS=destination["geometry"]["SRS"],
+                srcSRS=source["geometry"]["SRS"],
+                geometryType=destination["geometry"]["type"],
+                layerName=destination["name"],
+                accessMode="overwrite",
+                # optional settings
+                SQLStatement=destination.get("sql", ""),
+                callback=gdal.TermProgress,
+            )
+
+            # Compression if needed
+            if compress and destination_path:
+                if output_format == "ESRI Shapefile":
+                    files = [
+                        f"{destination_path[:-4]}.{suffix}"
+                        for suffix in ["shp", "prj", "shx", "dbf"]
+                    ]
+                    self.compress(f"{destination_path}.zip", *files, inplace=True)
+                else:
+                    self.compress(
+                        f"{destination_path}.zip", destination_path, inplace=inplace
+                    )
+
+        return wrapper
+
+    @translator
+    def postgres(
+        self, path: str, postgres_url: str, compress: bool = False, inplace=False
+    ):
         """
         This function will take in a configuration then send to a
         postgres database
         path: path of the configuration
         postgres_url: connection string for the destination database
+        compress: default to False because no files created when output to "PostgreSQL"
+        inplace: default to False because no compress = False by default
         """
-        # Initiate configuration
-        c = Config(path)
-        _, source, destination, _ = c.compute_parsed
-
-        # initiate source and destination datasets
         dstDS = postgres_source(postgres_url)
-        srcDS = generic_source(
-            path=source["url"]["gdalpath"],
-            options=source["options"],
-            fields=destination["fields"],
-        )
-
-        # Initiate translation
-        self.translate(dstDS, srcDS, "PostgreSQL", source, destination)
-        return True
+        return dstDS, "PostgreSQL", None, compress, inplace
 
-    def translate_csv(
+    @translator
+    def csv(
         self,
         path: str,
         compress: bool = False,
         inplace: bool = False,
-    ) -> bool:
+    ):
         """
         https://gdal.org/drivers/vector/pgdump.html
```

Comment:
Remove these URLs from all of the non-PG dump functions.

Reply:
I thought sqlalchemy is included in the pyproject.toml, I've added some new dependencies, you might need to run `poetry install` again

Reply:
Yup, I found it. Thanks!

Reply:
the layer creation option is different for all different kinds of file format, we don't have a custom way of handling it yet, ideally under destination options, we should have somethjing like
```yaml
options
   - csv
      - "this=that"
   - geojson
      - "that=this"
   - postgres
      - "OVERWRITE=YES"
```

Reply:
but this is lower priority, since if the file format is wrong, it would just throw a warning. we can handle them in future

Reply:
the reason I included these urls         https://gdal.org/drivers/vector/pgdump.html
is to help users to find the corresponding layercreationioptions, would need to update them at one point

Reply:
But why do we have the link for pgdump for all functions?

Reply:
oh I forgot to change them, the functions were copy and pasted